From: hasufell <julian.ospald@googlemail.com>
Date: Sun Apr 22 19:08:52 UTC 2012
Subject: various cmake fixes

fix hardcoded install paths for BINDIR and LIBDIR
make cg, cuda, glew, glut and openexr controllable and not automagic
add install-rules for headers (used by media-libs/devil)

--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -6,6 +6,13 @@
 
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
+# initial variables
+SET(GLUT TRUE CACHE BOOL "")
+SET(GLEW TRUE CACHE BOOL "")
+SET(CG TRUE CACHE BOOL "")
+SET(CUDA TRUE CACHE BOOL "")
+SET(OPENEXR TRUE CACHE BOOL "")
+
 # OpenGL
 INCLUDE(FindOpenGL)
 IF(OPENGL_FOUND)
@@ -15,13 +22,15 @@
 ENDIF(OPENGL_FOUND)
 
 # GLUT
-INCLUDE(${NV_CMAKE_DIR}/FindGLUT.cmake)
-#INCLUDE(FindGLUT)
-IF(GLUT_FOUND)
-	MESSAGE(STATUS "Looking for GLUT - found")
-ELSE(GLUT_FOUND)
-	MESSAGE(STATUS "Looking for GLUT - not found")
-ENDIF(GLUT_FOUND)
+IF(GLUT)
+	INCLUDE(${NV_CMAKE_DIR}/FindGLUT.cmake)
+	#INCLUDE(FindGLUT)
+	IF(GLUT_FOUND)
+		MESSAGE(STATUS "Looking for GLUT - found")
+	ELSE(GLUT_FOUND)
+		MESSAGE(STATUS "Looking for GLUT - not found")
+	ENDIF(GLUT_FOUND)
+ENDIF(GLUT)
 
 # DirectX
 INCLUDE(${NV_CMAKE_DIR}/FindDirectX.cmake)
@@ -32,29 +41,35 @@
 ENDIF(DX10_FOUND)
 
 # GLEW
-INCLUDE(${NV_CMAKE_DIR}/FindGLEW.cmake)
-IF(GLEW_FOUND)
-	MESSAGE(STATUS "Looking for GLEW - found")
-ELSE(GLEW_FOUND)
-	MESSAGE(STATUS "Looking for GLEW - not found")
-ENDIF(GLEW_FOUND)
+IF(GLEW)
+	INCLUDE(${NV_CMAKE_DIR}/FindGLEW.cmake)
+	IF(GLEW_FOUND)
+		MESSAGE(STATUS "Looking for GLEW - found")
+	ELSE(GLEW_FOUND)
+		MESSAGE(STATUS "Looking for GLEW - not found")
+	ENDIF(GLEW_FOUND)
+ENDIF(GLEW)
 
 # Cg
-INCLUDE(${NV_CMAKE_DIR}/FindCg.cmake)
-IF(CG_FOUND)
-	MESSAGE(STATUS "Looking for Cg - found")
-ELSE(CG_FOUND)
-	MESSAGE(STATUS "Looking for Cg - not found")
-ENDIF(CG_FOUND)
+IF(CG)
+	INCLUDE(${NV_CMAKE_DIR}/FindCg.cmake)
+	IF(CG_FOUND)
+		MESSAGE(STATUS "Looking for Cg - found")
+	ELSE(CG_FOUND)
+		MESSAGE(STATUS "Looking for Cg - not found")
+	ENDIF(CG_FOUND)
+ENDIF(CG)
 
 # CUDA
-INCLUDE(${NV_CMAKE_DIR}/FindCUDA.cmake)
-IF(CUDA_FOUND)
-	SET(HAVE_CUDA ${CUDA_FOUND} CACHE BOOL "Set to TRUE if CUDA is found, FALSE otherwise")
-	MESSAGE(STATUS "Looking for CUDA - found")
-ELSE(CUDA_FOUND)
-	MESSAGE(STATUS "Looking for CUDA - not found")
-ENDIF(CUDA_FOUND)
+IF(CUDA)
+	INCLUDE(${NV_CMAKE_DIR}/FindCUDA.cmake)
+	IF(CUDA_FOUND)
+		SET(HAVE_CUDA ${CUDA_FOUND} CACHE BOOL "Set to TRUE if CUDA is found, FALSE otherwise")
+		MESSAGE(STATUS "Looking for CUDA - found")
+	ELSE(CUDA_FOUND)
+		MESSAGE(STATUS "Looking for CUDA - not found")
+	ENDIF(CUDA_FOUND)
+ENDIF(CUDA)
 
 # Maya
 INCLUDE(${NV_CMAKE_DIR}/FindMaya.cmake)
@@ -93,13 +108,15 @@
 ENDIF(TIFF_FOUND)
 
 # OpenEXR
-INCLUDE(${NV_CMAKE_DIR}/FindOpenEXR.cmake)
-IF(OPENEXR_FOUND)
-	SET(HAVE_OPENEXR ${OPENEXR_FOUND} CACHE BOOL "Set to TRUE if OpenEXR is found, FALSE otherwise")
-	MESSAGE(STATUS "Looking for OpenEXR - found")
-ELSE(OPENEXR_FOUND)
-	MESSAGE(STATUS "Looking for OpenEXR - not found")
-ENDIF(OPENEXR_FOUND)
+IF(OPENEXR)
+	INCLUDE(${NV_CMAKE_DIR}/FindOpenEXR.cmake)
+	IF(OPENEXR_FOUND)
+		SET(HAVE_OPENEXR ${OPENEXR_FOUND} CACHE BOOL "Set to TRUE if OpenEXR is found, FALSE otherwise")
+		MESSAGE(STATUS "Looking for OpenEXR - found")
+	ELSE(OPENEXR_FOUND)
+		MESSAGE(STATUS "Looking for OpenEXR - not found")
+	ENDIF(OPENEXR_FOUND)
+ENDIF(OPENEXR)
 
 # Qt
 FIND_PACKAGE(Qt4)
@@ -121,3 +138,4 @@
 
 #INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/nvconfig.h DESTINATION include)
 
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/nvconfig.h DESTINATION include)
--- src/nvtt/CMakeLists.txt
+++ src/nvtt/CMakeLists.txt
@@ -53,13 +53,11 @@
 TARGET_LINK_LIBRARIES(nvtt ${LIBS} nvcore nvmath nvimage squish)
 
 INSTALL(TARGETS nvtt 
-	RUNTIME DESTINATION bin
-	LIBRARY DESTINATION lib
-	ARCHIVE DESTINATION lib/static)
-
-INSTALL(FILES nvtt.h DESTINATION include/nvtt)
-
+	RUNTIME DESTINATION ${BINDIR}
+	LIBRARY DESTINATION ${LIBDIR}
+	ARCHIVE DESTINATION ${LIBDIR})
 
+INSTALL(DIRECTORY . DESTINATION include/nvtt FILES_MATCHING PATTERN "*.h")
 
 # test executables
 ADD_EXECUTABLE(nvcompress tools/compress.cpp tools/cmdline.h)
--- src/nvmath/CMakeLists.txt
+++ src/nvmath/CMakeLists.txt
@@ -28,6 +28,8 @@
 TARGET_LINK_LIBRARIES(nvmath ${LIBS} nvcore)
 
 INSTALL(TARGETS nvmath
-	RUNTIME DESTINATION bin
-	LIBRARY DESTINATION lib
-	ARCHIVE DESTINATION lib/static)
+	RUNTIME DESTINATION ${BINDIR}
+	LIBRARY DESTINATION ${LIBDIR}
+	ARCHIVE DESTINATION ${LIBDIR})
+
+INSTALL(DIRECTORY . DESTINATION include/nvmath FILES_MATCHING PATTERN "*.h")
--- src/nvimage/CMakeLists.txt
+++ src/nvimage/CMakeLists.txt
@@ -62,7 +62,8 @@
 TARGET_LINK_LIBRARIES(nvimage ${LIBS} nvcore nvmath posh)
 
 INSTALL(TARGETS nvimage
-	RUNTIME DESTINATION bin
-	LIBRARY DESTINATION lib
-	ARCHIVE DESTINATION lib/static)
+	RUNTIME DESTINATION ${BINDIR}
+	LIBRARY DESTINATION ${LIBDIR} 
+	ARCHIVE DESTINATION ${LIBDIR})
 
+INSTALL(DIRECTORY . DESTINATION include/nvimage FILES_MATCHING PATTERN "*.h")
--- src/nvcore/CMakeLists.txt
+++ src/nvcore/CMakeLists.txt
@@ -42,6 +42,8 @@
 TARGET_LINK_LIBRARIES(nvcore ${LIBS})
 
 INSTALL(TARGETS nvcore
-	RUNTIME DESTINATION bin
-	LIBRARY DESTINATION lib
-	ARCHIVE DESTINATION lib/static)
+	RUNTIME DESTINATION ${BINDIR}
+	LIBRARY DESTINATION ${LIBDIR}
+	ARCHIVE DESTINATION ${LIBDIR})
+
+INSTALL(DIRECTORY . DESTINATION include/nvcore FILES_MATCHING PATTERN "*.h")
